### PR TITLE
Add get parameters directly

### LINKED ISSUE #6686: Add feature: get parameters (io)
In the training process, it is sometimes necessary to monitor the change of the parameters. The parameters become larger, the parameters become smaller or the parameters are constant. 
So, we need an interface to get parameters directly.

### PR DESCRIPTION
**Pull request by chengduoZH**
fix https://github.com/PaddlePaddle/Paddle/issues/6686
related https://github.com/PaddlePaddle/Paddle/issues/6645

### CODE REVIEW

File: python/paddle/v2/fluid/io.py [hunk 1]
```diff
@@ -278,3 +278,55 @@ def get_parameter_value_by_name(name, executor, program=None):
...
+            main_program = default_main_program()
+        if not isinstance(main_program, Program):
+            raise TypeError("program should be as Program type or None")
+
+        return get_vars(
+            executor, vars=filter(predicate, main_program.list_vars()))
+    else:
+        fetch_list = []
+        get_program = Program()
+        block = get_program.global_block()
```
Comment (dzhwinter) [inline python/paddle/v2/fluid/io.py]: Not every variable exists in global block. Only the persistable ones store in global block.

File: python/paddle/v2/fluid/io.py [hunk 2]
```diff
@@ -278,3 +278,55 @@ def get_parameter_value_by_name(name, executor, program=None):
...
+        for var in vars:
```
Comment (dzhwinter) [inline python/paddle/v2/fluid/io.py]: You can use the `vars() or all_parameters()` interface in `Block`.

**Comment by dzhwinter** [comment]
Providing an interface for getting parameter is great. But there is one big flaw in this PR, after the program execution,  you can just find the variable value in the scope based name identifier. We do not need to copy them into a separate program and execute it. 

In addition, last time I have a discussion on this topic with @reyoung .  I believe that it's better if we can return an class(or maybe decorator sugar) called `expression` instead of `variable`. 

```python
class Expression(object):
@property
def params(self):
     """
     return the contained parameters
     """
     self._params = []

def eval(self, default_executor):
    """
    let current variable as target, run main_program and return the result
    """
    
     
# usage
fc = layers.fc(...)
parameters = fc.params
```

Then the user can touch the parameter easily as their will.

**Comment by chengduoZH** [comment]
> We do not need to copy them into a separate program and execute it.

The current way of implementation is really a little bad. This is just to meet the current requirements, without considering the performance and other aspects. 
We need a convenient and efficient interface.

**Comment by luotao1** [comment]
感谢您给PaddlePaddle贡献代码。由于Paddle V1/V2版本已不再维护，相关代码也已从develop分支上删除，因此关闭您的PR，欢迎您向Paddle最新版-Fluid贡献代码。
Thanks for contributing to PaddlePaddle! Since V1/V2 will not be maintained anymore, and related codes have been deleted from develop branch as well, we close this PR. Welcome to contribute to Fluid——the latest version of PaddlePaddle.

**Pull request closed without merging**